src/framework/types/entity.ts

```diff
@@ -19,4 +21,49 @@ type EntityPropertyValue =
   | number[]
   | boolean
   | undefined
-  | null;
+  | null
+  | EntityRawData[];
```

Comment:
This is a tad yucky but was needed get the `_rawData` type to play nicely. tbh I'm considering just making `EntityPropertyValue` an `any` type.

Reply:
Can we make it essentially "not an object"?

Reply:
@charlieduong94 This makes sense to me tbh. Why do you say it's `yucky`?

Reply:
I mainly say this is yucky because it doesn't quite fulfill the constraints I want.

This allows for all additional fields (non core entity properties) to contain an `EntityRawData` array. From a typescript "loose" object perspective, it makes sense, but I don't want users to think that they can values they really can't.